### DISCUSSION TITLE
Neumann boundary condition Stokesflow

### DISCUSSION THREAD

**Discussion by VictordHaas** [PMEAL/OpenPNM#2680] [Q&A]
Dear OpenPNM Team,

I have a question about applying a Neumann boundary condition on the Stokesflow algorithm. I have previously only worked with applying a uniform discharge pressure (as shown in your examples), but this leads to unrealistic flow distributions in my current application: coupling multiple networks, which have a stark pore size gradient over their thickness direction, in their main flow direction. The top of the porous domain is home to large pores while the bottom is home to small pores, and there is no pore size gradient in the length direction. 

For the inlet network, I get the expected behavior over the whole domain except for close to the coupling face (outlet), with the expected behavior being that the flow rate (and velocity) is greatest in the large pores and that there is minimal cross-flow over the thickness of the porous domain towards the small pore size part. Just prior to the coupling face, however, the liquid starts dispersing over the whole thickness of the domain due to the application of a 0 [Pa] discharge pressure over the whole discharge face. 

Is there a possibility to assign an outlet Neumann boundary condition for the (internal) pores at this coupling face, or would the system be undetermined in this case? What I am thus searching for is to apply a certain rate on the outlet boundary pores equal to the net inflow rate of the internal network pores to which these are connected.

**Comment by jgostick** [comment]
I am having a bit of trouble picturing your exact situation, but I get the rough idea. My first thought is that OpenPNM enforces a mass balance, and flow goes where it goes, and that's that. However, I should ask what you've got for 'boundary pores'.  Have you added pores specificaly for this, or are you using internal pores to the set the outlet BCs?  If the former, are these all isolated from each other and only connecting to one internal pore each?  If not then there can be 'cross-talk' between the boundary pores since the conductance values are not physical.
